components/Header.tsx

```diff
@@ -0,0 +1,86 @@
+import { useRef, useState, useEffect, useCallback } from 'react';
+import { disableBodyScroll, enableBodyScroll } from 'body-scroll-lock';
+import { Theme } from 'types/Base';
+import { TorchboxLogo } from './TorchboxLogo';
+import styles from '/styles/Header.module.scss';
+
+import { MobileMenuButton, MobileNav } from './MobileNav';
+import { DesktopNav } from './DesktopNav';
+
+const NAVIGATION_LINKS = [
+    {
+        title: 'Design + build products',
+        url: 'https://torchbox.com/digital-products/',
+    },
+    {
+        title: 'Wagtail services',
+        url: 'https://torchbox.com/wagtail-cms/',
+    },
+    {
+        title: 'Digital marketing',
+        url: 'https://torchbox.com/digital-marketing/',
+    },
+    { title: 'Careers', url: '/careers/', isCareersSiteInternalLink: true },
+    {
+        title: 'Blog',
+        url: 'https://torchbox.com/blog/',
+    },
+    {
+        title: 'Events',
+        url: 'https://torchbox.com/events/',
+    },
+    {
+        title: 'Team',
+        url: 'https://torchbox.com/team/',
+    },
+];
+
+type HeaderProps = {
+    jobsAvailable?: number;
+    theme: Theme;
+};
+
+export const Header = ({ theme, jobsAvailable = 0 }: HeaderProps) => {
+    const [isOpen, setIsOpen] = useState(false);
+    const mobileNavRef = useRef<HTMLDivElement>(null);
+
+    const toggleMobileMenu = () => {
+        const mobileNav = mobileNavRef.current as HTMLElement;
+
+        if (isOpen) {
+            setIsOpen(false);
+            enableBodyScroll(mobileNav);
+        } else {
+            setIsOpen(true);
+            disableBodyScroll(mobileNav);
+        }
+    };
+
+    const themeClass = theme === 'DARK' ? 'themeDark' : 'themeLight';
+
+    return (
+        <div className={[styles.container, themeClass].join(' ')}>
```

Comment:
Set theme in the Layout component, not here: theme should be accessible everywhere in the document.

Reply:
I think you could just do string concatination here `${styles.container} ${themeClass}`, I remember suggesting this before but I feel like that was related to an already existing array of string values. These variables look to already be strings so creating an array shouldn't be neccassary.

You could also look into a useContext hook for keeping the theme value in sync.

Reply:
I've been swapping between different forms of contactination in this project, I think I will add the above suggestion for two classes throughout my project, and use the join for three or more classes.

Reply:
UseContext for the theme value looks interesting, I will have to learn more about this. As we have SCSS modules I think it's nicer to keep all the styling within the SCSS files, so we don't have to update the classes or styles conditionally with React.